stanford/roles/cluster/defaults/main.yml

```diff
@@ -181,6 +195,32 @@ CLUSTER_SECURITY_GROUP_RULES:
       to_port: 80
       cidr_ip: '0.0.0.0/0'
 CLUSTER_ELBS:
+  build:
+    - name: "{{ COMMON_DEPLOYMENT }}-elb-build-public"
+      health_check:
+        ping_protocol: http
+        ping_path: '/cc.xml'
+        ping_port: 80
+        response_timeout: 5  # seconds
+        interval: 30  # seconds
+        unhealthy_threshold: 2  # attempts
+        healthy_threshold: 10  # attempts
+      listeners:
+        - protocol: https
+          load_balancer_port: 443
+          instance_port: 80
+          instance_protocol: http
+          ssl_certificate_id: "{{ DEPLOYMENT_SSL_CERTIFICATE_ID }}"
+      scheme: internet-facing
+      group:
+        name: "{{ COMMON_DEPLOYMENT }}-group-build-elb-public"
+        rules:
+          - proto: tcp
+            from_port: 443
+            to_port: 443
+            cidr_ip: '0.0.0.0/0'
+      subnets:
+        - "{{ DEPLOYMENT_SUBNET_PUBLIC }}"
```

Comment:
FWIW, much of that change (`s/nat/public`) already exists on master [1][2];
we were already calling that subnet `public` and already placing public ELBs within.
The renaming that you see in this PR is me paying down technical debt and synchronizing the AWS tags to match the code itself.

I'm admittedly not clear on what you're being confused by, nor do I follow what you're suggesting.

> Given the amount of time it has taken me to eyeball this PR

Can you clarify what exactly you're having trouble with here?

I had considered this PR to be straight-forward and non-controversial.
Just like every other cluster we've configured to-date, it only includes:
- configuration for:
    - the subnet
    - the elb
    - the security groups
- a new playbook to invoke the cluster

The only 2/3 "unrelated" lines are to create 2 new variables which fix a bug (elbs going to the wrong subnet) and making it easier to extend (explained below).

> if [...] code can be split out to make it more autonomous

As we've discussed f2f, a "cluster" in this EDC architecture provides the infrastructure for a cluster/box/service to operate. These tasks, performed by `roles/cluster/tasks/main.yml` include:
- create a subnet
- add the subnet to a route table
- optionally, create an ELB w/ its own security group
- create a security group for the cluster itself
- if an ec2-backed cluster, create an ec2 instance.

These tasks, unless noted otherwise, are all required.

It's not clear to me what can be simplified; are you seeing something that I'm not?
Excluding `debug` statements (which probably should be removed at this point anyway),
the entire task file is under 100 LoC.

What are you trying to optimize?

I'm open to suggestions, but I'll need some more info on what exactly you're hoping to solve here.

[1] https://github.com/Stanford-Online/configuration/blob/master/stanford/roles/cluster/defaults/main.yml#L24
[2] https://github.com/Stanford-Online/configuration/blob/master/stanford/roles/cluster/defaults/main.yml#L89

Reply:
Sorry, I should have been more explicit.

My understanding is that this pr contains 2 functional changes:
- Adds a cluster for build, aka jenkins.
- Changes the subnet attached to the LB for all clusters that have an LB.

It's the second item that requires considerable perspicacity to trace though because it involves looking at each of the route tables for each of the LB subnets and comparing it to the deployment 'instance' subnet. Isn't that correct?

My, admittedly, oblique reference to autonomous code was that, in this case, it would have been easier if the code for each cluster was self-contained such that this change would only have affected the 'build' subnet. However, that's not an ideal solution, for obvious reasons, so I'm not advocating for that. Just that maybe there's some better way of architecting this.

Reply:
"However, for the public ELBs, the configured subnet was
ignored and we ended up updating it manually."

Well, that was my point; the subnets for these elbs will change. I didn't know that they had been updated manually but even when I look at them now they look odd. E.g. lagunita-elb-edxapp-public has 2 subnets that both have private RTs. It doesn't look to me as though the subnets for this ELB were edited.